enterprise_gateway/services/processproxies/k8s.py

```diff
@@ -227,7 +229,6 @@ def _determine_kernel_pod_name(self, **kwargs: dict[str, Any] | None) -> str:
         return pod_name
 
     def _determine_kernel_namespace(self, **kwargs: dict[str, Any] | None) -> str:
```

Comment:
(The following applies to lines L240-L254 below...)

1. Today, if the user is bringing their own namespace via `KERNEL_NAMESPACE` it is their responsibility to ensure proper operation _within_ the namespace.  With this PR, and if `EG_USE_REMOTE_CLUSTER` is True, I'm assuming the same requirement holds.  Are there any pieces of new functionality that would preclude users from bringing their own namespace (where that namespace resides in the remote cluster)?

2. Clearly, both `EG_SHARED_NAMESPACE` = True and `EG_USE_REMOTE_CLUSTER` = True conflict.  Perhaps we should check for `EG_SHARED_NAMESPACE` = True when obtaining the k8s client and either log and return the _local_ client or raise an exception?  

((Just to note, in `gateway_provisioners` the _default_ value for `EG_SHARED_NAMESPACE` is True to allow for easier deployments outside of something like since they are available to _any_ `jupyter_client` application.))

Reply:
1. Something I realize should be documented better is that remote cluster operation _requires_ the namespace to be already created before we launch a kernel there. This was mostly a security concern, as long as operations are isolated within a namespace in a remote cluster it's fine, but creating a namespace is a cluster level operation which can be scary for operators to enable. 

2. Agreed on this, we should log a warning then ignore it if `EG_USE_REMOTE_CLUSTER` is enabled